### DISCUSSION TITLE
AS keyword for aliases

### DISCUSSION THREAD

**Discussion by michal-bartak** [dbeaver/dbeaver#11674] [Q&A]
When inserting table names using autocomplete, DBeaver automatically generates an alias:
```
SELECT * FROM users.tab_users tu 
```
As you can see it lacks of AS keyword. Is there any option I missed to enable that? Or it's missing feature?

With regards

**Comment by ShadelessFox** [comment]
We don't include the `AS` clause for table aliases because it is not required and thus can be omitted.

**Comment by stuartramsay** [comment]
While it can be omitted, I'd consider to be best practising using it.
I will give an example. Lets say we have a 1000 line query to debug. We see a field aliased like a.name. We now have to look at all the joined tables to find which is aliased as 'a'
If we had used the AS keyword then its a simple case of searching for 'AS a' and we instantly find it.
Appreciate this is old, but its the only thing that annoys me about dbeaver and I landed here looking for a solution to have the option of including the AS keyword :)

**Reply by ged-yuko** [reply]
There is a setting for that actually.  You can find it down below on the `SQL Editor` --> `Code Completion` preference page (in the global preferences, or for specific connection settings):
![изображение](https://github.com/user-attachments/assets/31cf57f9-9be4-4556-a647-002da929eb03)

PS Note that i navigated to this setting by applying filter above the tree of the preference pages. 
There is no such filter in the `Edit Connection` dialog for per connection settings, but this preference page is easily locatable there as well.

**Reply by ged-yuko** [reply]
And, by the way, you can just Ctrl+Click alias `a` name, and the cursor will jump to the alias definition! 
Or use F4 hotkey, which stands for the `Go to definition` command by default. 

Here is a GIF animation demonstrating how it works:
![2025-03-2702-07-59-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/e2dd72cc-9494-4c21-9219-e5c462c4549e)

Note that it works for aliases and computed columns of subqueries with recently introduced Semantic analysis enabled. Largre queries might be tricky to analyze, so it would also be a good test of a new function! 🙃

**Reply by stuartramsay** [reply]
Everyday is a school day! thank you :)
